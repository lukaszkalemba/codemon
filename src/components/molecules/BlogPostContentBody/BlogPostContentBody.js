import React from 'react';
import PropTypes from 'prop-types';
import BlogPostContentItem from 'components/atoms/BlogPostContentItem';
import S from './BlogPostContentBody.styles';

const BlogPostContentBody = ({ content }) => {
  return (
    <S.Content>
      {content.map(item => (
        <BlogPostContentItem key={item.key} content={item} />
      ))}
    </S.Content>
  );
};

BlogPostContentBody.propTypes = {
  content: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.shape({})]))
    .isRequired,
};

export default BlogPostContentBody;
