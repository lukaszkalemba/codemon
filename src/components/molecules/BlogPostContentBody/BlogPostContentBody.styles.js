import styled from 'styled-components';

const Content = styled.div`
  display: flex;
  flex-direction: column;
  margin-top: -55px;

  div:nth-child(4n) {
    text-align: right;
    align-self: flex-end;
  }

  div:nth-child(4n - 1) {
    text-align: right;
  }

  div:first-child {
    margin-top: -30px;

    ${({ theme }) => theme.media.laptop`
      margin-top: -50px;
    `}
  }

  div:last-child {
    margin-bottom: -20px;

    ${({ theme }) => theme.media.desktop`
      margin-bottom: 30px;
    `}
  }
`;

export default { Content };
