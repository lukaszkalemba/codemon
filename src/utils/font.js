export default {
  size: {
    xxxs: '1rem',
    xxs: '1.3rem',
    xs: '1.6rem',
    s: '2.2rem',
    m: '3rem',
    l: '4rem',
    xl: '5.4rem',
    xxl: '8.5rem',
    xxxl: '10rem',
  },

  weight: {
    regular: '400',
    medium: '500',
    bold: '700',
  },

  family: {
    montserrat: 'Montserrat',
  },
};
