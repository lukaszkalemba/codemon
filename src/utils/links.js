export default [
  {
    id: 0,
    path: '/projekty',
    name: 'projekty',
  },
  {
    id: 1,
    path: '/blog',
    name: 'blog',
  },
  {
    id: 2,
    path: '/wycena',
    name: 'wycena',
  },
];
