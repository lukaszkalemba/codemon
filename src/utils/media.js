export default {
  phone: '576px',
  tablet: '768px',
  desktop: '1150px',
  largeDesktop: '1600px',
};
