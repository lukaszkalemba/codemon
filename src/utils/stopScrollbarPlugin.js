import { ScrollbarPlugin } from 'smooth-scrollbar';

class stopScrollbarPlugin extends ScrollbarPlugin {
  static pluginName = 'stopScrollbar';

  static defaultOptions = {
    open: false,
  };

  transformDelta(delta) {
    return this.options.open ? { x: 0, y: 0 } : delta;
  }
}

export default stopScrollbarPlugin;
