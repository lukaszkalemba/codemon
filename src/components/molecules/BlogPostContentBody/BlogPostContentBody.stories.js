import React from 'react';
import { text, withKnobs } from '@storybook/addon-knobs';
import BlogPostContentBody from './BlogPostContentBody';

export const blogPostContentBody = () => (
  <BlogPostContentBody
    content={[
      <>
        <h2>
          {text(
            'First Blog Post Content Item Title',
            'Lorem ipsum dolor sit amet.'
          )}
        </h2>
      </>,
      <>
        <p>
          {text(
            'First Blog Post Content Item Description',
            'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Odit doloribus illum, aspernatur nulla id nihil suscipit consequuntur dolores expedita similique eveniet ex ullam, est libero beatae quam, placeat accusamus et. Lorem ipsum, dolor sit amet consectetur adipisicing elit. Odit doloribus illum, aspernatur nulla id nihil suscipit consequuntur dolores expedita similique eveniet ex ullam, est libero beatae quam, placeat accusamus et.'
          )}
        </p>
      </>,
      <>
        <h2>
          {text(
            'Second Blog Post Content Item Title',
            'Lorem ipsum dolor sit amet.'
          )}
        </h2>
      </>,
      <>
        <p>
          {text(
            'Second Blog Post Content Item Description',
            'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Odit doloribus illum, aspernatur nulla id nihil suscipit consequuntur dolores expedita similique eveniet ex ullam, est libero beatae quam, placeat accusamus et. Lorem ipsum, dolor sit amet consectetur adipisicing elit. Odit doloribus illum, aspernatur nulla id nihil suscipit consequuntur dolores expedita similique eveniet ex ullam, est libero beatae quam, placeat accusamus et.'
          )}
        </p>
      </>,
    ]}
  />
);

export default {
  component: BlogPostContentBody,
  decorators: [withKnobs],

  title: 'Molecules|BlogPostContentBody',
};
