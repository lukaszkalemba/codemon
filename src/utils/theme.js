import color from 'utils/color';
import font from 'utils/font';
import media from 'utils/media';
import zindex from 'utils/zindex';

export default {
  color,
  font,
  media,
  zindex,
};
