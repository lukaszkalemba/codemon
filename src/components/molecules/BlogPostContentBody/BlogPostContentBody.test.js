import React from 'react';
import { render } from 'utils/test-utils';
import '__mocks__/intersectionObserver';
import BlogPostContentBody from './BlogPostContentBody';

describe('<BlogPostContentBody />', () => {
  it('renders correctly and matches snapshot', () => {
    const fakeContent = [
      <div key="0" />,
      <div key="1" />,
      <div key="2" />,
      <div key="3" />,
      <div key="4" />,
    ];
    const { container } = render(<BlogPostContentBody content={fakeContent} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
