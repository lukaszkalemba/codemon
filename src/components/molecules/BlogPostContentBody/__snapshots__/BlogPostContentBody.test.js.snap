// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogPostContentBody /> renders correctly and matches snapshot 1`] = `
<div
  class="sc-AxirZ hhfYiK"
>
  <div
    class="sc-AxjAm epiZDW"
    style="opacity: 0; transform: translateY(100px) translateZ(0);"
  >
    <div />
  </div>
  <div
    class="sc-AxjAm epiZDW"
    style="opacity: 0; transform: translateY(100px) translateZ(0);"
  >
    <div />
  </div>
  <div
    class="sc-AxjAm epiZDW"
    style="opacity: 0; transform: translateY(100px) translateZ(0);"
  >
    <div />
  </div>
  <div
    class="sc-AxjAm epiZDW"
    style="opacity: 0; transform: translateY(100px) translateZ(0);"
  >
    <div />
  </div>
  <div
    class="sc-AxjAm epiZDW"
    style="opacity: 0; transform: translateY(100px) translateZ(0);"
  >
    <div />
  </div>
</div>
`;
